### ISSUE TITLE
0.3.3 doesn't work with with Om 0.8.0 and react 0.12.2

### ISSUE THREAD

**Issue by mstang** [racehub/om-bootstrap#42]
Tested it, reverted back to 11.1.  Om 0.8.0 downloads 0.12.2.  My HTML references 0.11.1.  Tried to use 0.12.2, but that didn't work.  Not sure how long this bastardization combination will continue.  I guess at some point I will have to stop upgrading Om or om-bootstrap.

**Comment by lostman** [comment]
You can make it work but some changes are required. `om-bootstrap` explicitly depends on `om-0.7.1`, so add this to your `project.clj`:
```
[racehub/om-bootstrap "0.3.3" :exclusions [om]]
[org.om/om "0.8.0"]
```
Also, in `react-0.12.2` `isValidComponent` has been renamed to `isValidElement`. If you're planning to use advanced optimizations then I suggest adding a local copy of `om-bootstrap` to `checkouts/` and replacing the two occurrences of `isValidComponent` in `modal.cljs` and `util.cljs`.

Perhaps `om-bootstrap` could change the dependency to `[om "0.7.3" :scope "provided"]`?

**Comment by sritchie** [comment]
that's a good call. I didn't realize that David would change the org name in his dependency, which kills dependency resolution. I'll make that change today.

**Comment by sritchie** [comment]
Okay, I've just pushed `0.3.4-SNAPSHOT`. Can you guys try and let me know if it works? Here's the change:

https://github.com/racehub/om-bootstrap/pull/44

**Comment by sritchie** [comment]
@mstang, I still haven't gotten around to fixing that issue with the latest Om. Here it is: https://github.com/racehub/om-bootstrap/pull/40 You might take a look at that if it's urgent.

**Comment by sritchie** [comment]
Now that Om 0.8.0 is out, once we fix that issue I'm down to start pushing toward 0.4.0 on develop. @lostman, I'd take a pull req fixing those two references and other changes if you want. Thanks!

**Comment by sritchie** [comment]
@mstang, @lostman, can you guys confirm that this works? Cutting a release now.

**Comment by sritchie** [comment]
Okay, releasing 0.3.4 now. Give that a try. Also, I'm finally getting around to upgrading this lib to 0.8.0 today! The clone-with-props bug should get fixed by tomorrow. Thanks for your patience, guys.

**Comment by lgrapenthin** [comment]
Awesome that you are still on to it :+1:  Thanks.

**Comment by mstang** [comment]
Thanks, looking forward to upgrading everything ;-)!

**Comment by lgrapenthin** [comment]
Is it now working with om 8? Our production app quickly stops rendering and spams react errors in the console. I am asking because if it is supposed to work, we'd schedule debugging our code against it.

**Comment by sritchie** [comment]
@lgrapenthin, I wasn't able to get to it before a weekend trip - I'm REALLY hoping I can do this today and get the 0.8.0 bugs ironed out. Thanks for hanging with me, guys.

**Comment by sritchie** [comment]
Okay, fixed! 0.4.0-SNAPSHOT should be good to go with 0.8.x.

**Comment by mstang** [comment]
Great! I am looking forward to testing it!

Thanks for all your hard work!

regards,

Mark
